lib/ts/recipe/emailpassword/components/features/emailVerification/emailVerification.tsx

```diff
@@ -0,0 +1,296 @@
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Imports.
+ */
+import * as React from "react";
+import { PureComponent, Fragment } from "react";
+import {
+    EmailVerificationProps,
+    onHandleEmailVerificationSuccessContext,
+    VerifyEmailAPIResponse,
+    SendVerifyEmailThemeResponse,
+    VerifyEmailLinkClickedThemeProps,
+    SendVerificationEmailAPIResponse,
+    VerifyEmailThemeResponse,
+    SendVerifyEmailThemeProps
+} from "../../../types";
+import EmailPassword from "../../../emailPassword";
+import { EmailVerificationScreenTheme, signOut } from "../../..";
+import { RequestJson } from "../../../../../types";
+import FeatureWrapper from "../../../../components/featureWrapper";
+
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import { DEFAULT_VERIFY_EMAIL_PATH, EMAIL_VERIFICATION_MODE, SUCCESS_ACTION } from "../../../constants";
+import { getWindowOrThrow, redirectToInApp } from "../../../../../utils";
+import { handleVerifyEmailAPI, handleSendVerifyEmailAPI } from "./api";
+import Session from "../../../../session/session";
+import SuperTokens from "../../../../../superTokens";
+
+/*
+ * Component.
+ */
+
+class EmailVerification extends PureComponent<EmailVerificationProps, { token: string }> {
+    /*
+     * Constructor.
+     */
+    constructor(props: EmailVerificationProps) {
+        super(props);
+
+        const urlParams = new URLSearchParams(getWindowOrThrow().location.search);
+        let token = urlParams.get("token");
+        if (token === null) {
+            token = "";
+        }
+
+        this.state = {
+            token
+        };
+    }
+
+    /*
+     * Methods.
+     */
+    getRecipeInstanceOrThrow = (): EmailPassword => {
+        let instance;
+        if (this.props.__internal !== undefined && this.props.__internal.instance !== undefined) {
+            instance = this.props.__internal.instance;
+        } else {
+            instance = EmailPassword.getInstanceOrThrow();
+        }
+        return instance;
+    };
+
+    verifyEmail = async (): Promise<VerifyEmailThemeResponse> => {
+        return await handleVerifyEmailAPI(
+            this.getRecipeInstanceOrThrow().getRecipeId(),
+            this.onCallVerifyEmailAPI,
+            this.state.token
+        );
+    };
+
+    onEmailVerifiedSuccess = async (): Promise<void> => {
+        await this.onHandleSuccess({
+            action: SUCCESS_ACTION.EMAIL_VERIFIED_SUCCESSFUL
+        });
+    };
+
+    sendVerifyEmail = async (): Promise<SendVerifyEmailThemeResponse> => {
```

Comment:
`sendVerifyEmail` is used in the `componentDidMount` method and in the theme so it's useful to have a wrapper for it. `handleVerifyEmailAPI` was created to extract the API logic to  new files to reduce the size of each features.

The only place where I decided to not create an `api.ts` file is for `EmailPasswordAuth` `isEmailVerifiedAPI` because the wrapper code is really tiny: 

```
    isEmailVerifiedAPI = async (): Promise<boolean> => {
        try {
            const response = await this.onCallIsEmailVerifiedAPI({
                rid: this.getRecipeInstanceOrThrow().getRecipeId()
            });
            return response.isVerified;
        } catch (e) {
            // In case of API failure, continue, do not break the application.
            return false;
        }
    };
    ```